pkg/controllers/state/cluster.go

```diff
@@ -392,25 +383,14 @@ func (c *Cluster) cleanupMachine(name string) {
 }
 
 func (c *Cluster) newStateFromNode(ctx context.Context, node *v1.Node, oldNode *StateNode) (*StateNode, error) {
+	var n *StateNode
 	if oldNode == nil {
-		oldNode = NewNode()
-	}
-	n := &StateNode{
-		Node:                node,
-		Machine:             oldNode.Machine,
-		inflightAllocatable: oldNode.inflightAllocatable,
-		inflightCapacity:    oldNode.inflightCapacity,
-		startupTaints:       oldNode.startupTaints,
-		daemonSetRequests:   map[types.NamespacedName]v1.ResourceList{},
-		daemonSetLimits:     map[types.NamespacedName]v1.ResourceList{},
-		podRequests:         map[types.NamespacedName]v1.ResourceList{},
-		podLimits:           map[types.NamespacedName]v1.ResourceList{},
-		hostPortUsage:       scheduling.NewHostPortUsage(),
-		volumeUsage:         scheduling.NewVolumeUsage(),
-		volumeLimits:        scheduling.VolumeCount{},
-		markedForDeletion:   oldNode.markedForDeletion,
-		nominatedUntil:      oldNode.nominatedUntil,
+		n = NewNode()
+	} else {
+		n = oldNode.DeepCopy()
```

Comment:
Similar to above, but now it copies multiple maps it doesn't need to and then clears them.